src/Cassandra/Mapping/MapperFactory.cs

```diff
@@ -497,7 +497,13 @@ private Expression GetExpressionToGetColumnValueFromRow(ParameterExpression row,
             {
                 // Invoke the converter function on getValueT (taking into account whether it's a static method):
                 //     converter(row.GetValue<T>(columnIndex));
-                convertedValue = Expression.Call(converter.Target == null ? null : Expression.Constant(converter.Target), GetMethod(converter), getValueT);
+                convertedValue = 
+                    Expression.Convert(
```

Comment:
This is what fixes [CSHARP-710](https://datastax-oss.atlassian.net/browse/CSHARP-710). The object returned can be assigned to the target property but the types don't necessarily match so a cast is needed. 

For example, the target property might be `ICollection` or `IList` and the default converter will return a `List`. Without a cast, the `Expression.Condition()` call will fail because `List != IList`. 
The same cast is applied in the `TryGetCreateEmptyCollectionExpression` method for `IDictionary`, `ISet` and other collection interfaces:

```csharp
// Handle ICollection<T>, IList<T>, and IEnumerable<T>
if (openGenericType == typeof (ICollection<>) || openGenericType == typeof (IList<>) || openGenericType == typeof (IEnumerable<>))
{
	// The driver uses List so we'll use that as well
	Type listType = typeof (List<>).MakeGenericType(pocoDestType.GetTypeInfo().GetGenericArguments());

	// (... IList<T> or ICollection<T> or IEnumerable<T> ...) new List<T>();
	createEmptyCollection = Expression.Convert(Expression.New(listType), pocoDestType);
	return true;
}
```

Reply:
oh, makes sense, thanks for the detailed explanation!